tracer/src/Datadog.Trace/Agent/Transports/HttpClientResponse.cs

```diff
@@ -19,16 +20,41 @@ internal class HttpClientResponse : IApiResponse
         public HttpClientResponse(HttpResponseMessage response)
         {
             _response = response;
-
-            var encoding = _response.Content?.Headers?.ContentEncoding?.FirstOrDefault();
-            ContentEncoding = !string.IsNullOrEmpty(encoding) ? Encoding.GetEncoding(encoding) : Encoding.UTF8;
         }
 
         public int StatusCode => (int)_response.StatusCode;
 
         public long ContentLength => _response.Content.Headers.ContentLength ?? -1;
 
-        public Encoding ContentEncoding { get; }
+        public string ContentEncodingHeader => string.Join(',', _response.Content.Headers.ContentEncoding);
+
+        public string ContentTypeHeader => _response.Content.Headers.ContentType?.ToString();
+
+        public ContentEncodingType GetContentEncodingType() =>
+            _response.Content.Headers.ContentEncoding.Count switch
+            {
+                0 => ContentEncodingType.None,
+                1 => ApiResponseExtensions.GetContentEncodingType(_response.Content.Headers.ContentEncoding.First()),
```

Comment:
Nit-picking, but it feels weird to call a non-extension method that lives in `ApiResponseExtensions` from here. Also that `ApiResponseExtensions` lives in a file called `IApiResponse.cs`, which I would expect to contain only that interface. Should we try to reorg the code a little? 😅 (maybe later)

Reply:
> (maybe later)

😺